### DISCUSSION TITLE
Typescript template literal types causing an unexpected highlighting

### DISCUSSION THREAD

**Discussion by 9oelM** [github-linguist/linguist#5436] [Classification] [answered]
I wanted to open an issue but I'm not really sure if I could open an issue directly after reading the guideline, so i'm just posting it here. Also checked if there are existing issues/discussions/PRs - whatnot.

Since TS@4.1, it started supporting [template literal types](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#template-literal-types) and I've become a lover of it.

And it seems that it breaks(?) highlighting of the file, like this:
![image](https://user-images.githubusercontent.com/22465806/123530463-96501000-d735-11eb-975f-3931e8a52cd6.png)
You can have a look at the [code here](https://github.com/9oelM/youtube-lite/blob/dev/packages/app/src/utilities/redux-async/asyncTypes.ts), or just write the code yourself and test it out too because it's very easy to produce. The code just needs to contain a template literal type and some more codes below it.

You can see that before the template literal type appears, the highlighting goes ok, but as soon as it reaches the block with it, it seems to have that plain sky blue color for all of the codes. 

So.. I guess it's something to be fixed?

**Comment by 9oelM** [comment, marked as answer]
Oh. I think I need to report it to https://github.com/Microsoft/TypeScript-TmLanguage!

**Reply by lildude** [reply]
Normally that would be the place and is for most consumers of Linguist, but not GitHub. GitHub actually overrides Linguist and uses the tree-sitter grammar from https://github.com/tree-sitter/tree-sitter-typescript for TypeScript's syntax highlighting.

You can see the Microsoft grammar used by Linguist does appear to correctly handle your file when you view it using the grammar tester (doesn't work for tree-sitter grammars) [here](https://github-lightshow.herokuapp.com/?utf8=✓&scope=source.ts&grammar_format=auto&grammar_url=&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2F9oelM%2Fyoutube-lite%2Fblob%2Fdev%2Fpackages%2Fapp%2Fsrc%2Futilities%2Fredux-async%2FasyncTypes.ts&code=)

**Reply by lildude** [reply]
Looks like there's already an issue for it https://github.com/tree-sitter/tree-sitter-typescript/issues/136 😄

**Reply by 9oelM** [reply]
wooh yes. thank you for the pointer!
